src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -3207,15 +3232,20 @@ public String toString() {
     /**
      * Generates the next unique thread id.
      */
-    private static synchronized long nextThreadID() {
-        if (lastThreadID == Long.MAX_VALUE) {
-            if (logger.isLoggable(Level.FINER))
-                logger.finer("Resetting the Id count");
-            lastThreadID = 1;
-        } else {
-            lastThreadID++;
+    private static long nextThreadID() {
+        LOCK.lock();
+        try {
+            if (lastThreadID == Long.MAX_VALUE) {
+                if (logger.isLoggable(Level.FINER))
+                    logger.finer("Resetting the Id count");
+                lastThreadID = 1;
+            } else {
+                lastThreadID++;
+            }
+            return lastThreadID;
+        } finally {
+            LOCK.unlock();
```

Comment:
All of this could be tidied up with `AtomicLong#updateAndGet`

Reply:
Fair enough.  I did this in another commit.